src/config.js

```diff
@@ -0,0 +1,91 @@
+import request from 'request';
+import promisify from './promisify';
+import ConsultantError from './consultant-error';
+import properties from './properties';
+import fetchIdentifier from './identifier';
+import {parseKey, keyApplies} from './key-parser';
+import atob from 'atob';
+
+const loadConfig = async (consulHost, prefix, service) => {
+	const prefixUri = prefix ? `${prefix}/` : '';
+	const response = await promisify(cb => request({
```

Comment:
I could. But then in a few weeks/months I have another older lib I want to use with callbacks, and I'd need to revert everything again. I'd argue this is quite readable and abstracting it away further will only make it overly specific.